### DISCUSSION TITLE
possibility to create models as code

### DISCUSSION THREAD

**Discussion by SlitiBrahim** [pocketbase/pocketbase#3082] [Q&A]
Hey!

Is there a way to implement the models (tables) as code?

For example here I've lost my pb_data directory on my local dev setup and I've lost the model definitions. Is there a way to implement the models as code so at least only the rows are lost but not the schemas?

Thank you

**Comment by ganigeorgiev** [comment]
You can use migrations to keep track of your collections configuration or to pragmatically create collections:

- (Go) https://pocketbase.io/docs/go-migrations/
- (JS) https://pocketbase.io/docs/js-migrations/

**Comment by SlitiBrahim** [comment]
Thanks a lot! you rock 🔥
